gadopt/utility.py

```diff
@@ -475,3 +476,56 @@ def node_coordinates(function):
     return [
         Function(func_space).interpolate(coords).dat.data for coords in mesh_coords
     ]
+
+
+def interpolate_1d_profile(function: Function, one_d_filename: str, cartesian: bool):
+    """
+    Assign a one-dimensional profile to a Function `function` from a file.
+
+    The function reads a one-dimensional profile (e.g., viscosity) together with the
+    radius input for the profile, from a file, broadcasts the two arrays to all
+    processes, and then interpolates this array onto the function space of `function`.
+
+    Args:
+        function (Function): The function onto which the 1D profile will be assigned.
+        one_d_filename (str): The path to the file containing the 1D radial profile.
+        cartesian (bool): Is the upward direction along z/y(cartesian=True) or radial (cartesian=False)
+
+    Returns:
+        None: It directly interpolates the input function `function`.
+
+    Note:
+        - Note the cartesian flag
+        - This is designed to read a file with one process and distribute in parallel with MPI.
+        - The input file should contain an array of radius and an array of values, seperated by comma.
+    """
+    # find the mesh
+    mesh = extract_unique_domain(function)
+
+    # read the input file
+    if mesh.comm.rank == 0:
+        # The root process reads the file
+        rshl, visc = np.loadtxt(one_d_filename, unpack=True, delimiter=",")
+    else:
+        visc = None
+        rshl = None
+
+    # Broadcast the entire 'visc' array to all processes
+    visc = mesh.comm.bcast(visc, root=0)
+    # Similarly, broadcast 'rshl' if needed (assuming all processes need it)
+    rshl = mesh.comm.bcast(rshl, root=0)
+
+    # Coordinates
+    X = SpatialCoordinate(mesh)
+
+    if cartesian:
+        # if in a two-dimensional problem: y otherwise z is the direction
+        upward_coord = X[1] if len(X) == 2 else X[2]
+    else:
+        upward_coord = sqrt(X**2)
```

Comment:
Don't we have `vertical_component` to do this already?

Reply:
Correct.